syntax = "proto2";

package st;

// option go_package = "st.proto.ai/onboard/planning.assist/proto/external_command_status";

import "modules/cnoa_pnc/planning/proto/lite_common.proto";

import "modules/cnoa_pnc/planning/proto/autonomy_state.proto";
import "modules/cnoa_pnc/planning/proto/driving_style.proto";
import "modules/cnoa_pnc/planning/proto/alc.proto";
import "modules/cnoa_pnc/planning/proto/timestamp.proto";

message BlinkerOverrideStatusProto {
  optional bool on = 1;
}

message EnableFeatureStatusProto {
  optional bool traffic_light_stopping_is_enabled = 3;
  optional bool lc_objects_is_enabled = 4;
  optional bool pull_over_is_enabled = 5;

  reserved 1, 2;
}

// Inherited from PlannerTeleopStatusProto
message PlannerExternalCommandStatusProto {
  optional LiteHeader header = 1;

  optional BlinkerOverrideStatusProto left_blinker_override_status = 3;
  optional BlinkerOverrideStatusProto right_blinker_override_status = 4;
  optional BlinkerOverrideStatusProto emergency_blinker_override_status = 10;
  optional EnableFeatureStatusProto enable_feature_status = 6;
  optional double brake_to_stop = 12;
  optional double requested_stop_point_route_s = 7 [deprecated = true];
  optional double requested_forced_brake = 9 [deprecated = true];
  optional bool override_door_open = 13;
  optional bool enable_stop_polyline_stopping = 28;

  // optional QLCCState lcc_state = 14;
  optional ALCState alc_state = 15;
  // optional QACCState acc_state = 21;

  optional DriverAction.LaneChangeCommand lane_change_cmd = 22;
  optional st.planning.Timestamp plc_prepare_start_time = 34;

  optional float cruising_speed_limit = 18;
  optional LaneChangeStyle lane_change_style = 20;

  optional double current_lane_width = 23;
  optional double current_lane_length = 36;
  optional double current_lane_average_curvature_radius = 35;
  optional double distance_to_toll = 29;
  optional double distance_to_traffic_light = 30;
  optional bool is_av_overlap_boundary = 25;
  optional bool is_valid_both_side_boundary = 26;
  optional bool is_av_in_emergency_lane = 40;
  optional bool is_route_lane_change_fail = 27;
  optional bool lane_path_lost = 39;

  // optional planning.ExternalRouterCommand planning.to_router_command = 33;
  optional bool noa_need_lane_change_confirmation = 37;
  optional bool alc_confirmation = 38;

  // Next id: 41.
  reserved 2, 5, 8, 11, 16, 19, 24, 31, 32;
}